### PR TITLE
[ATH-1114] - Create Account

### PR DESCRIPTION
**Pull request by meganduffy**
User can now register by providing their first name, last name, unique email and password.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -41,6 +44,8 @@ group :development do
   # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
   gem 'spring'
   gem 'spring-watcher-listen', '~> 2.0.0'
+  gem 'shoulda-matchers'
+  gem 'rails-controller-testing'
```
Comment (jamesfairbairn-wpengine) [inline Gemfile]: I'm not sure, but I don't think you need to add this to both `:development, :test` and `test`. I think you could just add them to the `:test` one. Might need to check that though.

File: app/helpers/users_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
+end
```
Comment (jamesfairbairn-wpengine) [inline app/helpers/users_helper.rb]: Unless you need this class it is empty, so you currently don't really need it.

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class EmailValidator < ActiveModel::EachValidator
```
Comment (jamesfairbairn-wpengine) [inline app/models/user.rb]: You should really only have one class per file. You can move validators to `app/validators` see [here](https://stackoverflow.com/questions/42253003/where-should-i-put-the-custom-validators-in-rails-5)

File: spec/controllers/users_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+
+  describe "GET #new" do
```
Comment (jamesfairbairn-wpengine) [inline spec/controllers/users_controller_spec.rb]: If you aren't going to write tests for the `new` endpoint, then it's probably a good idea to just remove this block for now.

File: spec/helpers/users_helper_spec.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# require 'rails_helper'
```
Comment (jamesfairbairn-wpengine) [inline spec/helpers/users_helper_spec.rb]: This file is completely commented out, so shouldn't really be included in source control

File: spec/controllers/users_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+  end
+
+  describe "POST #create" do
+    context "with correct params" do
+      let(:parameters) { { user: { first_name: 'Megan', last_name: 'Duffy',
+                                   email: 'unique@email.com', password: 'hard2guess' } } }
+      it "redirects to homepage" do
+        post :create, params: parameters
+        expect(response).to redirect_to "/"
```
Comment (jamesfairbairn-wpengine) [inline spec/controllers/users_controller_spec.rb]: I think you could replace `"/"` with `root_path`

File: spec/requests/users_request_spec.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (jamesfairbairn-wpengine) [inline spec/requests/users_request_spec.rb]: Again, this file is basically empty, so we don't really need to include it.

**Comment by jamesfairbairn-wpengine** [comment]
Few minor comments. Functionality-wise, it looks good. Also, did we decide to leave off using RuboCop for the moment? Just wondering, as if we didn't, there are a few violations at the moment.

File: app/models/user.rb (hunk 3 above)
Comment (meganduffy) [reply app/models/user.rb]: This is very useful, thank you!

**Comment by meganduffy** [comment]
We are not using RuboCop currently, maybe it's something that could be revisited soon.

I implemented a few changes and removed those blank files, going to merge it in now!

**Pull request merged**